Lib/ast.py

```diff
@@ -968,8 +969,24 @@ def visit_Try(self, node):
             with self.block():
                 self.traverse(node.finalbody)
 
+    def visit_Try(self, node):
+        try:
+            prev_in_try_star = self._in_try_star
```

Comment:
This is pretty silly, but I like to have the "save" step outside the try block, so a failure to save doesn't trigger a failure in the finally block as well:
```suggestion
        prev_in_try_star = self._in_try_star
        try:
```
And ditto below.